facebook/facebook-ios-sdk
# Spm corekit

Thanks for proposing a pull request!

To help us review the request, please complete the following:


## Pull Request Details

Added a `Package.swift` manifest. 
Updated imports to be non-modular for non-cocoapod setups. 
Updated dynamic framework loader to compile with arc. - This should be checked. Not 100% sure I did not cause a regression.
Added a directory of symlinks to public header files - this was to get around a limitation in the SPM api where they will only accepts a path to header files (forces them to be located in the same directory) and not a list of header files.

## Test Plan

Test Plan:

1. Checkout branch
2. Open the package itself with >= Xcode 11 by opening `Package.swift` instead of the project and verify that it builds
3. Open a new project with >= Xcode 11 and add a new swift package. For the Package Repository field put the path to your checkout. Ex: `file:///Users/$whoami/facebook-ios-sdk` then specify the branch as this one, `spm-corekit`. This sample project should build and you should be able to consume the ObjC and Swift code by adding `import FBSDKCoreKit` and `import FacebookCore` respectively.

Should still be able to build the `FBSDKCoreKit` and `FBSDKCoreKitSwift` targets via xcode, CocoaPods, any other existing channels.
